roles/hosts-file/tasks/main.yml

```diff
@@ -0,0 +1,50 @@
+---
+
+  - name: logic flags | compute
+    set_fact:
+        do_clear: >
+            {{ hosts_file_clear_ansible_group != "" }}
+
+  - name: hosts file | probe
+    command: cat /etc/hosts
+    register: hosts_probe
+    when: "{{ do_clear|bool }}"
+
+  - name: hosts file | filter
+    set_fact:
+        filtered_hosts: >
+            {{ hosts_probe.stdout|hosts_file_filter(
+                groups[hosts_file_clear_ansible_group],
+                domain,
+                hostvars,
+                extra_clear_entries,
+                ansible_hostname,
+                ansible_nodename,
+                ansible_fqdn) }}
+    when: "{{ do_clear|bool }}"
+
+  - name: hosts file | clear
+    copy:
+        content: "{{ filtered_hosts }}"
+        dest: /etc/hosts
+    when: "{{ do_clear|bool }}"
+
+  - name: hosts file | amend
+    lineinfile:
+        dest: /etc/hosts
+        regexp: "^{{ item[0]|regex_escape() }}"
+        line: >
+            {{ item[0] }} {{ item[1]|join(" ") }}
+        state: >
+            {{ "present" if item[0] in ("127.0.0.1", "::1") else state }}
+    with_items: >
+        {{  groups[hosts_file_ansible_group]
+          | hosts_file_entries(hosts_file_net_interface,
+                               domain,
+                               hostvars,
+                               extra_entries,
+                               local_entries|bool,
+                               ansible_hostname,
+                               ansible_nodename,
+                               ansible_fqdn) }}
+
```

Comment:
Not sure why hosts_file_ansible_group is not getting picked up correctly.  Here is the playbook i am running with the inventory vagrant generates:

hosts-file.yml:
```yaml
- hosts: all
  roles:
    - role: hosts-file
      hosts_file_ansible_group: all
      domain: cluster.dev
```

inventory:
```yaml
# Generated by Vagrant

name ansible_ssh_host=127.0.0.1 ansible_ssh_port=2220 ansible_ssh_private_key_file=/home/kotfic/kitware/projects/gobig/src/gobig/.vagrant/machines/name/virtualbox/private_key
data-01 ansible_ssh_host=127.0.0.1 ansible_ssh_port=2221 ansible_ssh_private_key_file=/home/kotfic/kitware/projects/gobig/src/gobig/.vagrant/machines/data-01/virtualbox/private_key

[namenodes]
name

[datanodes]
data-01

[all:children]
namenodes
datanodes


```